src/braket/jobs/hybrid_job.py

```diff
@@ -0,0 +1,397 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+import functools
+import importlib.util
+import inspect
+import re
+import shutil
+import sys
+import tempfile
+import warnings
+from collections.abc import Callable, Iterable
+from logging import Logger, getLogger
+from pathlib import Path
+from types import ModuleType
+from typing import Any
+
+import cloudpickle
+
+from braket.aws.aws_session import AwsSession
+from braket.jobs._entry_point_template import run_entry_point, symlink_input_data
+from braket.jobs.config import (
+    CheckpointConfig,
+    InstanceConfig,
+    OutputDataConfig,
+    S3DataSourceConfig,
+    StoppingCondition,
+)
+from braket.jobs.image_uris import Framework, built_in_images, retrieve_image
+from braket.jobs.quantum_job import QuantumJob
+from braket.jobs.quantum_job_creation import _generate_default_job_name
+
+
+def hybrid_job(
+    *,
+    device: str | None,
+    include_modules: str | ModuleType | Iterable[str | ModuleType] = None,
+    dependencies: str | Path | None = None,
+    local: bool = False,
+    job_name: str | None = None,
+    image_uri: str | None = None,
+    input_data: str | dict | S3DataSourceConfig = None,
+    wait_until_complete: bool = False,
+    instance_config: InstanceConfig | None = None,
+    distribution: str | None = None,
+    copy_checkpoints_from_job: str | None = None,
+    checkpoint_config: CheckpointConfig | None = None,
+    role_arn: str | None = None,
+    stopping_condition: StoppingCondition | None = None,
+    output_data_config: OutputDataConfig | None = None,
+    aws_session: AwsSession | None = None,
+    tags: dict[str, str] = None,
+    logger: Logger = getLogger(__name__),
+) -> Callable:
+    """Creates a job by invoking the Braket CreateJob API.
```

Comment:
This isn't valid if local=True. Could we update to make it unambiguous?

Reply:
Good point!